src/sirtextstyle.c

```diff
@@ -203,15 +205,57 @@ bool _sir_setcolormode(sir_colormode mode) {
         /* when the color mode changes, it's necessary to regenerate the text styles
          * we're holding. for example in the case of downgrading color modes, the
          * styles in memory could be incompatible with the new mode. */
-        if (!_sir_resettextstyles()) {
+        if (!_sir_resettextstyles()) { // GCOVR_EXCL_START
             _sir_selflog("error: failed to reset text styles!");
             _SIR_UNLOCK_SECTION(SIRMI_TEXTSTYLE);
             return false;
-        }
+        } // GCOVR_EXCL_STOP
     } else {
         _sir_selflog("skipped superfluous update of color mode: %"PRId32, mode);
     }
     _SIR_UNLOCK_SECTION(SIRMI_TEXTSTYLE);
 
     return true;
 }
+#else /* SIR_NO_TEXT_STYLING */
+bool _sir_settextstyle(sir_level level, const sir_textstyle* style) {
+    SIR_UNUSED(level);
+    SIR_UNUSED(style);
+    return false;
+}
+
+const char* _sir_gettextstyle(sir_level level) { // GCOVR_EXCL_START
```

Comment:
This is a legitimate use of GCOVR_EXCL_START. This and the default case in a switch are pretty much the only kind that are though.